orttraining/orttraining/python/training/onnxblock/checkpoint_utils.py

```diff
@@ -1,26 +1,30 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License.
-# checkpoint_utils.py
+
+import os
+from typing import List, Tuple, Union
+
+import onnx
 
 from onnxruntime.capi._pybind_state import get_model_after_loading_checkpoint as _internal_load_checkpoint_to_model
 from onnxruntime.capi._pybind_state import save_checkpoint as _internal_save_checkpoint
 
 
-def save_checkpoint(parameters, path_to_checkpoint):
+def save_checkpoint(
+    parameters: Tuple[List[onnx.TensorProto], List[onnx.TensorProto]], path_to_checkpoint: Union[str, os.PathLike]
+) -> None:
     """Saves the parameters to the checkpoint directory path_to_checkpoint."""
 
     if parameters is None:
         raise RuntimeError("No checkpoint parameters provided.")
 
-    # TODO: use Parameter class to pass information to backend
-    # Serialize the parameters and save the checkpoint
     trainable_params, non_trainable_params = parameters
     trainable_params = [param.SerializeToString() for param in trainable_params]
     non_trainable_params = [param.SerializeToString() for param in non_trainable_params]
     _internal_save_checkpoint(trainable_params, non_trainable_params, path_to_checkpoint)
```

Comment:
in the PR description "- As a consequence, only args requiring grad are saved to the checkpoint file."

is that valid?

Reply:
OK. May I know why "non_trainable_params" is empty?  and maybe non trainable params should also be specfied by users?

Reply:
We don't know how to distinguish non trainable params from constant initializers. So, we make both these as part of model proto. Which is why non trainable params are empty here.

Reply:
I made the change so users must pass in the frozen params in case they know this information.